# Estruturas de decisão

<span style="color: red;"><b>Todos os exercícios são feitos partindo-se do pressuposto de que todas as entradas são dadas de forma correta. Casos limite não mencionados no enunciado não são abordados porque não fazem parte do exercício.</b></span>

#### 1. Faça um Programa que peça dois números e imprima o maior deles.

In [146]:
numberA = int(input('Digite um número:'))
numberB = int(input('Digite um segundo número:'))   
if numberA > numberB:
    print(numberA)
else:
    print(numberB)

4


#### 2. Faça um Programa que peça um valor e mostre na tela se o valor é positivo ou negativo.

In [ ]:
number = int(input('Digite um número:'))
if number > 0:
    print('O número é positivo')
else:
    print('O número é negativo')

O número é negativo


#### 3. Faça um Programa que verifique o estado civil de uma pessoa. Se a letra digitada é "C" (Casado), "S" (Solteiro), "D" (Divorciado), "V" (Viúvo) ou "O" (outros). Conforme a letra escrita pelo usuário seu programa deve escrever o estado civil, exemplo:

Usuário digita: C

Seu programa deve responder:
C - Casado

In [ ]:
estado_civil = input('Qual o seu estado civil?')
if estado_civil == 'c' or estado_civil == 'C':
    print('C - Casado')
elif estado_civil == 's' or estado_civil == 'S':
    print('S - Solteiro')
elif estado_civil == 'v' or estado_civil == 'V':
    print('V - Viúvo')
elif estado_civil == 'd' or estado_civil == 'D':
    print('D - Divorciado')
else:
    print('O - Outros')

S - Solteiro


#### 4. Faça um Programa que verifique se o e-mail digitado faz parte dos e-mails de spam.

In [ ]:
emails_spam = "fulano@gmail.com,beltrano@gmail.com,ciclano@gmail.com"


In [ ]:
email = input('Digite o seu melhor email:')
if email in emails_spam:
    print('Isso é um spam')
else:
    print('email válido')

Isso é um spam


#### 5. Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:

##### A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
##### A mensagem "Reprovado", se a média for menor do que sete;
##### A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [ ]:
nota1 = float(input('Qual foi a primeira nota?'))
nota2 = float(input('Qual foi a segunda nota?'))
media = (nota1 + nota2)/2
if media == 10:
    print('Aprovado com Distinção! Média 10.0!')
elif media >= 7:
    print('Aprovado com media {}' .format(media))
else:
    print('Reprovado com mádia {}' . format(media))


Aprovado com media 8.895


#### 6. Faça um Programa que leia o orçamento de 3 empresas e mostre o maior deles.

In [ ]:
orcamento1 = 8000
orcamento2 = 7012
orcamento3 = 7400

if orcamento1 > orcamento2 and orcamento1 > orcamento3:
    print ('O maior orcamento é o 1')
elif orcamento2 > orcamento3 and orcamento2 > orcamento1:
    print('O orcamento 2 é o maior')
else:
    print('O orcamento 3 é o maior')

O maior orcamento é o 1


#### 7. Faça um Programa que leia três orçamentos e mostre o maior e o menor deles.

In [ ]:
if orcamento1 > orcamento2 and orcamento1 > orcamento3:
    print ('O maior orcamento é o 1')
elif orcamento2 > orcamento3 and orcamento2 > orcamento1:
    print('O orcamento 2 é o maior')
else:
    print('O orcamento 3 é o maior')

if orcamento1 < orcamento2 and orcamento1 < orcamento3:
    print ('O menor orcamento é o 1')
elif orcamento2 < orcamento3 and orcamento2 < orcamento1:
    print('O orcamento 2 é o menor')
else:
    print('O orcamento 3 é o menor')

O maior orcamento é o 1
O orcamento 2 é o menor


#### 8. Faça um programa que pergunte o preço de três produtos e informe qual produto você deve comprar, sabendo que a decisão é sempre pelo mais barato.

In [ ]:
preco_prod1 = float(input('Preco do produto 1: '))
preco_prod2 = float(input('Preco do produto 2: '))
preco_prod3 = float(input('Preco do produto 3: '))

if preco_prod1 < preco_prod2 and preco_prod1 < preco_prod3:
    print('O produto a ser adquirido é o produto 1 pois apresenta o menor preço {}' .format(preco_prod1))
elif preco_prod2 < preco_prod1 and preco_prod2 < preco_prod3:
    print('O produto a ser adquirido é o produto 2 pois apresenta o menor preço {}' .format(preco_prod2))
else:
    print('O produto a ser adquirido é o produto 3 {}' .format(preco_prod3))

O produto a ser adquirido é o produto 3 23.32


#### 9. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [ ]:
n1 = int(input('Digite o number 1:'))
n2 = int(input('Digite o number 2:'))
n3 = int(input('Digite o number 3:'))

if n1 > n2 and n1 > n3 and n2 > n3:
    print(n1, n2, n3)
elif n1 > n2 and n1 > n3 and n3 > n2:
    print(n1, n3, n2)
elif n2 > n1 and n2 > n3 and n1 > n3:
    print(n2, n1, n3)
elif n2> n1 and n2 > n3  and n3 > n1:
    print(n2, n3, n1)
elif n3> n1 and n3 > n2  and n2 >  n1:
    print(n3, n2, n1)
else:
    print(n3, n1, n2)    

4 3 2


#### 10. Faça um Programa que pergunte em que turno você estuda. Peça para digitar M-matutino ou V-Vespertino ou N- Noturno. Imprima a mensagem "Bom Dia!", "Boa Tarde!" ou "Boa Noite!" ou "Valor Inválido!", conforme o caso.

In [ ]:
turno = input('Em qual turno voce estuda? ')

if turno == 'M':
    print('Bom dia!')
elif turno == 'V':
    print('Boa tarde!')
elif turno == 'N':
    print('Boa noite!')
else: 
    print('valor inválido')

valor inválido


#### 11. As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contraram para desenvolver o programa que calculará os reajustes. Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:

##### salários até R\\$ 280,00 (incluindo) : aumento de 20% 

##### salários entre R\\$ 280,00 e R\\$ 700,00 : aumento de 15% 

##### salários entre R\\$ 700,00 e R\\$ 1500,00 : aumento de 10% 

##### salários de R\\$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado, informe na tela: 

###### o salário antes do reajuste;

###### o percentual de aumento aplicado;

###### o valor do aumento;

###### o novo salário, após o aumento.

###### Obs: Não vamos nos preocupar tanto com a formatação dos números (nº de casas decimais, por exemplo, veremos isso no próximo módulo)

In [ ]:
salario = float(input('Digite o seu salário:'))
if salario <= 280:
    print('O salário antes do reajuste era de R$ {}'.format(salario))
    print('O percentual de aumento será de 20%')
    print('O valor do aumento foi de R$ {}' .format(salario*.2))
    print('O novo salário é de R$ {}'.format(salario+(salario*.2)))
elif salario > 280 and salario <= 700:
    print('O salário antes do reajuste era de R$ {}'.format(salario))
    print('O percentual de aumento será de 15%')
    print('O valor do aumento foi de R$ {}' .format(salario*.15))
    print('O novo salário é de R$ {}'.format(salario+(salario*.15)))
elif salario > 700 and salario <= 1500:
    print('O salário antes do reajuste era de R$ {}'.format(salario))
    print('O percentual de aumento será de 10%')
    print('O valor do aumento foi de R$ {}' .format(salario*.10))
    print('O novo salário é de R$ {}'.format(salario+(salario*.10)))
else:
    print('O salário antes do reajuste era de R$ {}'.format(salario))
    print('O percentual de aumento será de 5%')
    print('O valor do aumento foi de R$ {}' .format(salario*.05))
    print('O novo salário é de R$ {}'.format(salario+(salario*.05)))

O salário antes do reajuste era de R$ 1250.0
O percentual de aumento será de 10%
O valor do aumento foi de R$ 125.0
O novo salário é de R$ 1375.0


#### 12 . Faça um programa para o cálculo de uma folha de pagamento, sabendo que os descontos são do Imposto de Renda, que depende do salário bruto (conforme tabela abaixo) e que o FGTS corresponde a 11% do Salário Bruto, mas não é descontado (é a empresa que deposita). O Salário Líquido corresponde ao Salário Bruto menos os descontos. O programa deverá pedir ao usuário o valor da sua hora e a quantidade de horas trabalhadas no mês.

Desconto do IR:<br>
Salário Bruto até 900 (inclusive) - isento<br>
Salário Bruto até 1500 (inclusive) - desconto de 5%<br>
Salário Bruto até 2500 (inclusive) - desconto de 10%<br>
Salário Bruto acima de 2500 - desconto de 20%<br>
Imprima na tela as informações, dispostas conforme o exemplo abaixo. No exemplo o valor da hora é 5 e a quantidade de hora é 220.

Salário Bruto: (5 * 220)        : R\\$ 1100,00<br>
(-) IR (5%)                     : R\\$   55,00<br>
(-) INSS ( 10%)                 : R\\$  110,00<br>
FGTS (11%)                      : R\\$  121,00<br>
Total de descontos              : R\\$  165,00<br>
Salário Liquido                 : R\\$  935,00<br>

###### Obs: Não vamos nos preocupar tanto com a formatação dos números (nº de casas decimais, por exemplo, veremos isso no próximo módulo)

In [ ]:
valor_hora = float(input('Digite o valor da hora de trabalho: '))
horas_trabalhadas_mes = float(input('Quantas horas trabalhou esse mês: '))
salario_bruto = float(valor_hora * horas_trabalhadas_mes)

if salario_bruto <= 900:
    percentual = 0
elif salario_bruto <= 1500:
    percentual = 0.05
elif salario_bruto <= 2500:
    percentual = 0.10
else: 
    percentual = 0.20

irpf = salario_bruto * percentual
inss = salario_bruto * 0.1
fgts = salario_bruto *0.11
    
print('O salário bruto é R$ {:.2f}' .format(salario_bruto))
print('(-) desconto de IR = {:.2f}% : R$ {:.2f}' .format((percentual*100),(irpf)))
print('(-) desconto de INSS = R$ {:.2f}' .format((inss)))
print('FGTS = R$ {:.2f}' .format((fgts)))

salario_liquido = salario_bruto - inss - irpf
print('O salario_liquido é R$ {:.2f}'.format(salario_liquido) )    

    

O salário bruto é R$ 4995.00
(-) desconto de IR = 20.00% : R$ 999.00
(-) desconto de INSS = R$ 499.50
FGTS = R$ 549.45
O salario_liquido é R$ 3496.50


#### 13. Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [ ]:
dia_da_semana =int(input())
if dia_da_semana == 1:
    print("Hoje é domingo!")
elif dia_da_semana == 2:
    print("Hoje é segunda-feira!")
elif dia_da_semana == 3:
    print("Hoje é terça-feira!")
elif dia_da_semana == 4:
    print("Hoje é quarta-feira!")
elif dia_da_semana == 5:
    print("Hoje é quinta-feira!")
elif dia_da_semana == 6:
    print("Hoje é sexta-feira!")
elif dia_da_semana == 7:
    print("Hoje é sábado!")
else:
    print("Valor inválido")

Valor inválido


#### 14. Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. Em seguida, mostre qual conceito o aluno teve. A atribuição de conceitos obedece à tabela abaixo:
<pre>
Média de Aproveitamento  Conceito
Entre 9.0 e 10.0        A
Entre 7.5 e 9.0         B
Entre 6.0 e 7.5         C
Entre 4.0 e 6.0         D
Entre 4.0 e zero        E
</pre>

In [ ]:
nota1 = float(input("Digite a nota 1: "))
nota2 = float(input("Digite a nota 2: "))
media = (nota1+nota2)/2

if media > 9 and media <= 10:
    print('O conceito do aluno é "A" e a sua média foi {:.2f}' .format(media))
elif media > 7.5 and media <= 9:
    print('O conceito do aluno é "B" e a sua média foi {:.2f}' .format(media))
elif media > 6 and media <= 7.5:
    print('O conceito do aluno é "C" e a sua média foi {:.2f}' .format(media))
elif media > 4 and media <= 6:
    print('O conceito do aluno é "D" e a sua média foi {:.2f}' .format(media))
elif media <= 4:
    print('O conceito do aluno é "E" e a sua média foi {:.2f}' .format(media))

O conceito do aluno é "D" e a sua média foi 5.50


#### 15. Você está construindo um calendário para controlar dias de trabalho a pedido do RH. Nessa construção, você vai precisar definir quais anos são bissextos e quais não são, para montar o calendário de forma correta. Faça um Programa que peça um número correspondente a um determinado ano e em seguida informe se este ano é ou não bissexto.
<pre>
Dica para determinar se um ano é bissexto: 
- São bissextos todos os anos múltiplos de 400, p.ex.: 1600, 2000, 2400, 2800...
- São bissextos todos os múltiplos de 4, exceto se for múltiplo de 100 mas não de 400, 
p.ex.: 1996, 2000, 2004, 2008, 2012, 2016, 2020, 2024, 2028...
- Não são bissextos todos os demais anos.<br>
ex1: 2004 é múltiplo de 4, mas não é múltiplo de 100, então é bissexto.
ex2: 2000 é múltiplo de 4, mas é múltiplo de 100, só que também é multiplo de 400, então é bissexto (porque todo ano múltiplo de 400 é bissexto, independente do resto).
ex3: 1900 é múltiplo de 4, é múltiplo de 100, mas não é múltiplo de 400, então não é bissexto

</pre>

Dica: lembre que: numero % 4 é o resto da divisão do número por 4, ex: 10 % 3 = 1 (já que 10/3 = 3 e resta 1)

In [ ]:
ano = int(input("Digite o ano:"))
if ano % 400 == 0:
    print("O ano digitado foi {} e esse é um ano bissexto" .format(ano))
elif ano % 4 == 0 and ano %100 != 0:
    print("O ano digitado foi {} e esse é um ano bissexto" .format(ano))    
else:
    print("O ano digitado foi {} e esse não é um ano bissexto" .format(ano))

O ano digitado foi 2027 e esse não é um ano bissexto


#### 16. Faça um Programa para leitura de três notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e presentar:
<pre>
A mensagem "Aprovado", se a média for maior ou igual a 7, com a respectiva média alcançada;
A mensagem "Reprovado", se a média for menor do que 7, com a respectiva média alcançada;
A mensagem "Aprovado com Distinção", se a média for igual a 10.
</pre>

In [ ]:
nota1 = float(input("Digite a nota1: "))
nota2 = float(input("Digite a nota2: "))
nota3 = float(input("Digite a nota3: "))

media = (nota1+nota2+nota3)/3

if media == 10:
    print("Aprovado com Distinção, média {:.1f}" .format(media) )
elif media >= 7:
    print("Aprovado com média {:.1f}" .format(media))
else:
    print("Reprovado com média {:.1f}" .format(media))
 

Reprovado com média 5.7


#### 17. João Papo-de-Pescador, homem de bem, comprou um microcomputador para controlar o rendimento diário de seu trabalho. Toda vez que ele traz um peso de peixes maior que o estabelecido pelo regulamento de pesca do estado de São Paulo (50 quilos) deve pagar uma multa de R$ 4,00 por quilo excedente. João precisa que você faça um programa que leia a variável peso (peso de peixes) e calcule o excesso. Gravar na variável excesso a quantidade de quilos além do limite e na variável multa o valor da multa que João deverá pagar. Imprima os dados do programa com as mensagens adequadas.

In [ ]:
peso = float(input("Digite o peso do pescado hoje: "))
multa = 4
if peso >= 50:
    excedente = (peso - 50)
    print("Houve um excedente de {} kg" .format(excedente))
    print("Para pescas com excedente, há uma multa de R$ {:.2f}" .format(multa*excedente))
else:
    print("Pesca dentro do permitido.")
     

Houve um excedente de 37.0 kg
Para pescas com excedente, há uma multa de R$ 148.00


#### 18. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.
<pre>
Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;
Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.
</pre>

Dica1: numero // 10 vai te dar como resposta a parte inteira da divisão do número por 10.<br>
Dica2: numero % 10 vai te dar o resto da divisão do número por 10.

In [ ]:
valor = int(input("Quanto deseja sacar?"))

n100 = 0
n50 = 0
n20 = 0
n10 = 0
n5 = 0
n1 = 0

if 10 <= valor <= 600:
    print("São necessárias: ")
    if valor >= 100:
        n100 = valor // 100
        print("{} cédulas de R$ 100" .format(n100))
    if valor % 100 >= 50:
        n50 = (valor - 100*n100)//50
        print("{} cédulas de R$ 50" .format(n50))
    if valor % 50 >= 20:
        n20 = (valor - 100*n100 - 50*n50)//20
        print("{} cédulas de R$ 20" .format(n20))
    if valor % 20 >= 10:
        n10 = (valor - 100*n100 - 50*n50 - 20*n20)//10
        print("{} cédulas de R$ 10" .format(n10))
    if valor % 10 >= 5:
        n5 = (valor - 100*n100 - 50*n50- 20*n20 - 10*n10)//5
        print("{} cédulas de R$ 5" .format(n5))
    if valor % 5 >= 1:
        n1 = (valor - 100*n100 - 50*n50 - 20*n20 - 10*n10 - 5*n5)
        print("{} cédulas de R$ 1" .format(n1))
else:
    print("Informe um valor entre R$ 10 e R$ 600")
    

São necessárias: 
1 cédulas de R$ 100
1 cédulas de R$ 50
1 cédulas de R$ 20


#### 19. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
<pre>
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?" 
O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".
</pre>

In [ ]:
p1 = 0
p2 = 0
p3 = 0
p4 = 0
p5 = 0

resposta = input("Telefonou para a vítima?")
if resposta == "sim":
    p1 = 1

resposta = input("Esteve no local do crime?")
if resposta == "sim":
    p2 = 1
resposta = input("Mora perto da vítima?")
if resposta == "sim":
    p3 = 1

resposta = input("Devia para a vítima?")
if resposta == "sim":
    p4 = 1
resposta = input("Já trabalhou com a vítima?")
if resposta == "sim":
    p5 = 1


classificacao = p1 + p2 + p3+ p4 +p5
if classificacao == 5:
    print("É o ASSASSINO!")
elif 3 <= classificacao <= 4:
    print("É CÚMPLICE!")
elif classificacao == 2:
    print("É SUSPEITO")
else:
    print("Liberado.")

Liberado.


#### 20. Um posto está vendendo combustíveis com a seguinte tabela de descontos:
<pre>
Álcool:
até 20 litros, desconto de 3% por litro
acima de 20 litros, desconto de 5% por litro
Gasolina:
até 20 litros, desconto de 4% por litro
acima de 20 litros, desconto de 6% por litro

Escreva um algoritmo que leia o número de litros vendidos, o tipo de combustível (codificado da seguinte forma: A-álcool, G-gasolina), calcule e imprima o valor a ser pago pelo cliente sabendo-se que o preço do litro da gasolina é R$ 2,50 o preço do litro do álcool é R$ 1,90.
</pre>

In [ ]:
tipo_comb = input("A ou G")
quantidade = float(input("Quantidade de combustível:"))
if tipo_comb == "A" and quantidade <=20:
    total = (4.94*quantidade) 
    desconto = total * 0.03
     
elif tipo_comb == "A" and quantidade >20:
    total = (4.94*quantidade) 
    desconto = total * 0.05
     
elif tipo_comb == "G" and quantidade <=20:
    total = (6.40*quantidade)
    desconto = total * 0.03
    
elif tipo_comb == "G" and quantidade >20:
    total = (6.40*quantidade)
    desconto = total * 0.05
print(f"Total de litros abastecidos: {quantidade:.2f}")
print(f"O desconto total foi de R$: {desconto:.2f} ")
print(f"Total a pagar R$: {total:.2f}") 

Total de litros abastecidos: 49.55
O desconto total foi de R$: 15.86 
Total a pagar R$: 317.12


#### 21. Uma fruteira está vendendo frutas com a seguinte tabela de preços:
<pre>
                      Até 5 Kg           Acima de 5 Kg
Morango         R$ 2,50 por Kg          R$ 2,20 por Kg
Maçã            R$ 1,80 por Kg          R$ 1,50 por Kg
Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.
</pre>

In [ ]:
peso_morango = float(input("Digite o peso do morango: "))
peso_maca = float(input("Digite o peso da maça: "))

if peso_morango > 5:
    preco_morango = 2.20 * peso_morango
else:
    preco_morango = 2.50 * peso_morango

if peso_maca > 5:
    preco_maca = 1.50 * peso_maca
else:
    preco_maca = 1.80 * peso_maca
    
preco_total_morango = preco_morango * peso_morango    
print(f"O total de morangos foi de {peso_morango} kg, preço total R$ {preco_morango:.2f}")

preco_total_maca = preco_maca * peso_maca  
print(f"O total de maças foi de {peso_maca} kg, preço total R$ {preco_maca:.2f}")

preco_total = preco_total_maca + preco_total_morango  
if peso_morango + peso_maca > 8 or preco_total > 25:
    preco_total = (preco_maca + preco_morango)*.9
print(f"Total a pagar: R$ {preco_total:.2f}")


O total de morangos foi de 7.69 kg, preço total R$ 16.92
O total de maças foi de 6.79 kg, preço total R$ 10.19
Total a pagar: R$ 24.39


#### 22. O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:
<pre>
                      Até 5 Kg           Acima de 5 Kg
File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg
Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente. Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra. Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal, contendo as informações da compra: tipo e quantidade de carne, preço total, tipo de pagamento, valor do desconto e valor a pagar.
</pre>

In [115]:
tipo_carne = input("Digite o tipo de carne: ")
peso_carne = float(input("Digita o peso da carne: "))
tipo_pagamento = input("Qual o tipo de pagamento?")    

if peso_carne < 5 and tipo_carne == 'File Duplo':
    preco_carne = 4.9
elif peso_carne >= 5 and tipo_carne == 'File Duplo':
    preco_carne = 5.9

elif peso_carne < 5 and tipo_carne == 'Alcatra':
    preco_carne = 5.9
elif peso_carne >= 5 and tipo_carne == 'Alcatra':
    preco_carne = 6.9

elif peso_carne < 5 and tipo_carne == 'Picanha':
    preco_carne = 6.9
else:
    preco_carne = 7.9 
    
print("_______________CUPOM FISCAL_______________")            
print(f"Tipo de Carne {tipo_carne}")
print(f"Peso da carne: {peso_carne:.2f} kg ")
print(f"Preço do quilo da carne: R$ {preco_carne:.2f}")  
preco_total = float(preco_carne * peso_carne)
if tipo_pagamento == 'Cartão':
    desconto = preco_total * .05
print(f"O desconto cartão Tabajara foi de R$ {desconto:.2f}")
print(f"Preço total com desconto Tabajara: R$ {preco_total - desconto:.2f}")

_______________CUPOM FISCAL_______________
Tipo de Carne Alcatra
Peso da carne: 11.25 kg 
Preço do quilo da carne: R$ 6.90
O desconto cartão Tabajara foi de R$ 3.88
Preço total com desconto Tabajara: R$ 73.74


#### 23. Faça um Programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 6 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R\\$ 80,00 ou em galões de 3,6 litros, que custam R\\$ 25,00.

##### Informe ao usuário as quantidades de tinta a serem compradas e os respectivos preços em 3 situações.

Dica: lembre dos operadores // e % mostrados em exercícios anteriores<br>
Dica1: numero // 10 vai te dar como resposta a parte inteira da divisão do número por 10.<br>
Dica2: numero % 10 vai te dar o resto da divisão do número por 10.

##### 1. Comprar apenas latas de 18 litros: (apenas latas inteiras)

In [147]:
area = float(input('Informe o tamanho da área a ser pintada em m²: '))
print(f"A área a ser pintada é de {area:.2f} m2")
litros_tinta = area / 6
print(f"Serão necessários {litros_tinta:.2f} litros de tinta")

custo_lata = 80

necessidade = litros_tinta // 18

if necessidade < 1:
    lata = 1
else:
    lata = necessidade + 1
          
 
print(f"Serão necessárias {lata:.1f} latas de tinta")
print(f"{lata} latas de 18 litros contém {lata * 18} litros de tinta, sobrarão {(lata * 18) - litros_tinta:.2f}")    
custo_total = lata * custo_lata  
print(f"O custo total é para a pintura será: R$ {custo_total:.2f}")


A área a ser pintada é de 65.00 m2
Serão necessários 10.83 litros de tinta
Serão necessárias 1.0 latas de tinta
1 latas de 18 litros contém 18 litros de tinta, sobrarão 7.17
O custo total é para a pintura será: R$ 80.00


##### 2. Comprar apenas galões de 3,6 litros: (apenas galoes inteiros)

In [148]:
area = float(input('Informe o tamanho da área a ser pintada em m²: '))
print(f"A área a ser pintada é de {area:.2f} m2")
litros_tinta = area / 6
print(f"Serão necessários {litros_tinta:.2f} litros de tinta")

custo_galao = 25

necessidade = litros_tinta // 3.6
if necessidade < 1:
    galao = 1
else:
    galao = necessidade + 1
    
 
print(f"Serão necessários {galao:.1f} galões de tinta")
print(f"{galao} galões de 3,6 litros contém {galao * 3.6} litros de tinta, sobrarão {(galao * 3.6) - litros_tinta:.2f}")    
custo_total = galao * custo_galao 
print(f"O custo total é para a pintura será: R$ {custo_total:.2f}")

A área a ser pintada é de 65.00 m2
Serão necessários 10.83 litros de tinta
Serão necessários 4.0 galões de tinta
4.0 galões de 3,6 litros contém 14.4 litros de tinta, sobrarão 3.57
O custo total é para a pintura será: R$ 100.00


##### 3. Misturar latas e galões, de forma que o desperdício de tinta seja menor. Acrescente 10% de folga e sempre arredonde os valores para cima, isto é, considere latas cheias.

O custo da lata é 80/18 = 4,44 R\\$/L

O custo do galão é 25/3,6 = 6,94 R\\$/L

A lata é mais econômica, então todas as latas inteiras que pudermos usar devemos comprar em latas. Se ficar faltando alguma coisa para completar devemos avaliar se é melhor comprar latas ou galões. Exemplo:

Se queremos comprar 90 litros. 5 latas dão exatamente 90 litros. Então devemos comprar tudo em latas.

Se queremos comprar 95 litros. 5 latas dão exatamente 90 litros. Então devemos comprar pelo menos 5 latas e avaliar o que falta, se estes últimos 5 litros valem mais apenas em latas ou galões.

Para os 5 litros faltantes precisamos de 2 galões que custam 50 reais no total. Ou de uma lata que custa 80 reais no total. Portanto, neste caso vale mais a pena usar 2 galões.

Se queremos comprar 107 litros. 5 latas dão exatamente 90 litros. Então devemos comprar pelo menos 5 latas e avaliar o que falta, se estes últimos 5 litros valem mais apenas em latas ou galões.

Para os 17 litros faltantes precisamos de 5 galões que custam 125 reais no total. Ou de uma lata que custa 80 reais no total. Portanto, neste caso vale mais a pena usar uma lata.

3 galões custam 75 reais, 4 galões custam 100 reais. Então, se for possível completar com até 3 galões escolhe-se galões. Qualquer quantidade maior que 3 galões, usa-se latas.

Podemos ir ao exercício:

In [181]:
area = float(input('Informe o tamanho da área a ser pintada em m²: '))
print(f"A área a ser pintada é de {area:.2f} m2")
litros_tinta = (area / 6)*1.1
print(f"Serão necessários {litros_tinta:.2f} litros de tinta")

latas = litros_tinta // 18
galoes = 0

tinta_faltante = litros_tinta - 18 * latas

if tinta_faltante > 0:
    if tinta_faltante % 3.6 > 0:
        faltante_galoes = tinta_faltante // 3.6 + 1
    else:
        faltante_galoes = tinta_faltante // 3.6
    if faltante_galoes <= 3:
        galoes = faltante_galoes
    else:
        galoes = 0
        latas = latas + 1

preco = latas * 80 + galoes * 25
print(f'São necessários {latas} latas e {galoes} galões. Custando R$ {preco} no total')
        
 

A área a ser pintada é de 220.00 m2
Serão necessários 40.33 litros de tinta
São necessários 2.0 latas e 2.0 galões. Custando R$ 210.0 no total
